### PR TITLE
Store again

### PR DESCRIPTION
**Pull request by tranma**
WIP 
porting psql and tests

### CODE REVIEW

File: lib/Synchronise/Document.hs [hunk 1]
```diff
@@ -41,13 +43,14 @@ data Document = Document
 
 makeLenses ''Document
 
-instance ToJSON Document where
-    toJSON = _documentContent
+$(deriveJSON defaultOptions ''Document)
```
Comment (thsutton) [inline lib/Synchronise/Document.hs]: This is probably wrong. What are we going to do with the JSON of a `Document` (as opposed to the `Value` which is its `_documentContent`)?

Comment (tranma) [inline lib/Synchronise/Document.hs]: The *Document functions need JSON instances for PostgreSQL. I dunno, what's the semantics of those functions?

Comment (thsutton) [inline lib/Synchronise/Document.hs]: I delete this and it still builds.

Comment (tranma) [inline lib/Synchronise/Document.hs]: probs leftover from when things are labelled
I'll clean up

Comment (tranma) [inline lib/Synchronise/Document.hs]:
```

lib/Synchronise/Store/PostgreSQL.hs:111:56-61:
    No instance for (ToJSON Synchronise.Document.Document)
      arising from a use of ‘toJSON’
    In the expression: toJSON doc
    In the third argument of ‘execute’, namely
      ‘(entity, ik', ik', entity, toJSON doc)’
    In a stmt of a 'do' block:
      _ <- execute conn sql (entity, ik', ik', entity, toJSON doc)

```
you didn't get this?

Comment (tranma) [inline lib/Synchronise/Document.hs]: oh, you mean revert the change.
